pennylane_lightning_gpu/src/simulator/StateVectorCudaBase.hpp

```diff
@@ -115,15 +124,25 @@ class StateVectorCudaBase : public StateVectorBase<Precision, Derived> {
                                    bool async = false) {
         PL_ABORT_IF_NOT(BaseType::getLength() == length,
                         "Sizes do not match for Host and GPU data");
-        if (!async) {
-            PL_CUDA_IS_SUCCESS(cudaMemcpy(data_, gpu_sv,
-                                          sizeof(CFP_t) * BaseType::getLength(),
-                                          cudaMemcpyDefault));
-        } else {
-            PL_CUDA_IS_SUCCESS(cudaMemcpyAsync(
-                data_, gpu_sv, sizeof(CFP_t) * BaseType::getLength(),
-                cudaMemcpyDeviceToDevice, stream_));
-        }
+        data_buffer_->CopyGpuDataToGpu(gpu_sv, length, async);
+    }
+    /**
+     * @brief Explicitly copy data from antoher GPU device memory block to this
```

Comment:
```suggestion
     * @brief Explicitly copy data from another GPU device memory block to this
```

Reply:
Nice work! I just pointing out a typo here, but you have my OK to merge after fixing that.

Reply:
Well spotted!